MinecraftClient.openScreen -> setScreen

Juuxel opened the pull request:
Closes #1597. The issue has the main reasoning, but I'd also mention that an error string within the method calls it `setScreen`.

This is also a semi-impactful rename, though less than #2511 so I don't think it needs to be delayed to after a release.

Also fixes a tiny javadoc error in `Item.Settings`.

Juuxel committed:
MinecraftClient.openScreen -> setScreen

Closes #1597.

Juuxel committed:
Fix javadoc error in Item.Settings

liach commented on mappings/net/minecraft/client/MinecraftClient.mapping:
```diff
@@ -200,12 +200,15 @@ CLASS net/minecraft/class_310 net/minecraft/client/MinecraftClient
-		COMMENT If the currently opened screen is {@code null} and player is dead then the death screen will be opened.
-		COMMENT Otherwise the currently open screen will be closed.
+		COMMENT If the currently opened screen is {@code null} and player is dead, then the death screen will be opened.
+		COMMENT Otherwise, the currently open screen will be closed.
```
I suggest replace `currently` with `previous` to make it more clear (not confuse current with the passed parameter)

Juuxel committed:
'currently opened' -> 'previous'

liach commented on mappings/net/minecraft/client/MinecraftClient.mapping:
```diff
@@ -204,8 +204,8 @@ CLASS net/minecraft/class_310 net/minecraft/client/MinecraftClient
 		COMMENT <p>If the screen being opened is {@code null} and the client is not in game, the title screen will be opened.
-		COMMENT If the currently opened screen is {@code null} and player is dead, then the death screen will be opened.
-		COMMENT Otherwise, the currently open screen will be closed.
+		COMMENT If the previous screen is {@code null} and the player is dead, then the death screen will be opened.
```
So the death screen would override the provided parameter screen when the player is dead, I assume?

Juuxel replied:
No, see the line above - this only applies when the new screen is null.

liach replied:
Hmm, so a table format would be better:
```
<p>If the new screen is {@code null}:
<ul>
<li>if the client is not in game, the title screen will be opened;</li>
<li>if the previous screen is {@code null} and the player is dead, then the death screen will be opened.</li>
</ul>
```

liach commented on mappings/net/minecraft/client/MinecraftClient.mapping:
```diff
@@ -200,12 +200,15 @@ CLASS net/minecraft/class_310 net/minecraft/client/MinecraftClient
+		COMMENT If the previous screen is {@code null} and the player is dead, then the death screen will be opened.
+		COMMENT Otherwise, the previous screen will be closed.
+		COMMENT
+		COMMENT <p>If there is an open screen when the current screen is changed, {@link Screen#removed()} will
```
Consider merging this and the previous sentence, both of which talk about screen removal.

Juuxel committed:
Improve jd

Juuxel commented:
@liach I changed the docs a bit again, and also fixed a factual error about the death screen.

Rendered:
![rendered JD](https://user-images.githubusercontent.com/6596629/123550000-03829600-d774-11eb-924e-ab10ea8ceba4.png)

liach commented:
> (The double @Nullable happens elsewhere too, not sure what causes it)

Fyi this is because the jetbrains nullable is both type-use and parameter capable. Hence, when mappingpoet generates `@Nullable Type value`, javadoc would interpret that nullable to be used for both cases, and hence it's rendered twice.

github-actions[bot] committed:
Merge branch '1.17.1-pre2' into set-screen

github-actions[bot] committed:
Merge branch '1.17.1-rc1' into set-screen

modmuss50 reviewed (approved):
I like this change, as this is qutie impactful I think its best we hold off on merging it untill we are also happy to merge: https://github.com/FabricMC/yarn/pull/2511

github-actions[bot] committed:
Merge branch '1.17.1' into set-screen
